### DISCUSSION TITLE
[Question][Hw1] Can I add test cases?

### DISCUSSION THREAD

**Discussion by ShinWonho** [prosyslab-classroom/cs524-program-analysis#403] [Q&A] [answered]
Because the number of given test cases is few, I want to add some more test cases.
Is changing the test/[Problem name]Test.ml files prohibited?

**Comment by hyunsukimsokcho** [comment, marked as answer]
One can change `test/*Test.ml` files as one wants. Still, note that we are not going to validate those newly added or changed test cases.

**Comment by ShinWonho** [comment]
Thank you!

**Comment by DreadPirateRobertz** [comment]
### miquella's take — Implementation path for Option 2

Agreed with radahn on dropdown + visual swatches. From the design system perspective:

**Color tokens already support this.** `sharedTokens.js` has our finish colors mapped — Espresso `#3A2518`, Sand/Natural `#E8D5B7`. We just need to extend the token set with specific wood finish values (Walnut, Cherry, Oak) and expose them through `designTokens.js`.

**Swatch component pattern exists.** `swatchPromoSection` in Home.js already renders color circles for promotional display. The pattern is: repeater → circle element → background-color from token → click handler. Adapting this for product options is minimal work.

**Accessibility concern with swatches:** Color alone isn't enough for WCAG AA. Each swatch needs:
- `aria-label` with the finish name (not just the color)
- A text label below/beside the swatch for color-blind users
- Selected state with a visible border (not just color change)
- `a11yHelpers.js` `announce()` call on selection: *"Selected: Espresso finish"*

**Out-of-stock UX:** Rather than tooltip (requires hover, fails on mobile), I'd suggest:
- Swatch shows with diagonal strikethrough line + reduced opacity (0.4)
- Tapping shows inline text: *"Espresso not available in Queen. Available in: Twin, Full"*
- This pattern works identically on mobile and desktop

**Mobile layout (320px):** Size dropdown full-width above, finish swatches as horizontal scroll row below. Swatch min-size 44px tap target per `a11yHelpers.js` specs.
